File: scala/hamming/src/main/scala/Hamming.scala
object Hamming {
  def distance(strand1: String, strand2: String): Option[Int] =
    strand1.length() == strand2.length() match {
      case false => None
      case true  => Some((strand1 zip strand2).count { case (a, b) => a != b })
    }
}

File: fsharp/hamming/Hamming.fs
module Hamming

let distance (strand1: string) (strand2: string) : int option =
    match strand1.Length = strand2.Length with
    | false -> None
    | true ->
        Some (Seq.zip strand1 strand2 |> (Seq.filter (fun (x, y) -> x <> y) >> Seq.length))
